### ISSUE TITLE
Initial Redux DevTools support ("v0.5")

### ISSUE THREAD

**Issue by markerikson** [RecordReplay/devtools#6106]
Add built-in support for showing the Redux DevTools as a panel in the UI similar to the React DevTools, and show action+state data captured from the app.

As part of this effort, document the integration approach needed so that we can reuse this approach for other framework devtools later.

Tasks:

- #6498 
- #6391
- #6132  
- #6505
- #6583
- #6584
- #6585

**Comment by markerikson** [comment]
Update, 2022-05-08:

We now have a "v0.5" implementation actually working!  The browser is correctly capturing Redux actions data from apps as Replay annotations.  (I initially didn't see this working on my machine, using an in-place update of the FF browser on Windows, but we have confirmed that fresh downloads of the browser work okay on both Win and Mac.)

The UI side currently just renders the entire Redux DevTools UI app component, with all the existing RDT UI pieces inside (actions list, "Action/State/Diff/Trace" tabs, etc).  This is all behind an "Experimental Settings" flag, so it will only try to show the "Redux" tab if you have that turned on (_and_ there are Redux annotations available).

However, I talked with @jasonLaster today about what we want for a full "v1" implementation, and we're going to take this in a different direction:

- We'll reuse the main RDT reducer logic, but in our Redux store
- Reuse/modify existing Replay UI panes for actions list / state tabs
- Current RDT UI will be a separate popout tab

Reasons:

We want to keep a common look and feel in Replay's UI. Also, we want to integrate things like the actions list with Replay's behavior - show arrow for current time location, jump to pause or stack trace, etc.

The existing RDT components can't do that as-is.

However, there's also value in having the entire existing Redux DevTools UI still, and we've already shown we can use that.

So, similar to our "Sourcemap Explorer" integration, we'll add a button that pops out the RDT UI as a totally separate browser tab from the main Replay UI.

I'm going to spend a week or so trying to do some cleanup on our codebase ( per #6552 ), and then circle back and start working on the revised Redux DevTools UI implementation.

Jason put up a very pseudo-code-y PR that doesn't at all run, but demonstrates the ideas of where we want to go, over in #6666 .
